### PR TITLE
configure lint-staged

### PR DESCRIPTION
**Pull request by benschac**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues,
Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

**Comment by benschac** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1048** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1048" target="_blank">Graphite</a> 
    * **PR #1049** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1049" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1049?utm_source=stack-comment).

File: .eslintignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 build
-node_modules
\ No newline at end of file
+node_modules
+.eslintrc.json
```
Comment (benschac) [inline .eslintignore:3]: I figure eslint is formatted and has a really strict schema so we could ignore it all together.

File: .eslintrc.json [hunk 2]
```diff
@@ -33,5 +33,13 @@
         ]
       }
     ]
-  }
+  },
+  "overrides": [
```
Comment (benschac) [inline .eslintrc.json:37]: I used overrides here, but I guess we could ignore package.json altogether too? Not sure how y'all feel about it.

File: package.json [hunk 3]
```diff
@@ -70,7 +70,7 @@
     "format": "prettier --write './**/*.{js,jsx,ts,tsx}'"
   },
   "lint-staged": {
-    "src/**/*.{js,jsx,ts,tsx,json,css}": [
+    "*.{js,jsx,ts,tsx,json,css,md}": [
```
Comment (benschac) [inline package.json:73]: `lint-staged` will now run on all markdown files, all files that aren't `node_modules`. Jacked this from the docs: https://github.com/okonet/lint-staged#automatically-fix-code-style-with-prettier-for-javascript-typescript-markdown-html-or-css

File: src/hooks/Poller.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ export function usePoller(
 
   // run at start too
   useEffect(
-    function() {
```
Comment (aeolianeth) [inline src/hooks/Poller.ts:12]: **suggestion**: This hook looks unused. Want to remove it?

**Review by aeolianeth — APPROVED** [review]
Sweet @benschac , this all LGTM. Just one suggestion about deleting a file.

**Review by aeolianeth — CHANGES_REQUESTED** [review]
.

**Comment by benschac** [comment]
> Sweet @benschac , this all LGTM. Just one suggestion about deleting a file.

Dope, I can create another PR (diff) addressing the issue.

**Pull request merged**